tests/agent_server/test_server_agent.py

```diff
@@ -485,6 +501,53 @@ async def verify_deployment_result():
     await resource_action_consistency_check()
 
 
+async def test_env_setting_wiring_to_autostarted_agent(
+    resource_container, environment, client, clienthelper, no_agent_backoff, async_finalizer, server
+):
+    """
+    Test that the AUTOSTART_AGENT_DEPLOY_INTERVAL and AUTOSTART_AGENT_REPAIR_INTERVAL
+    env settings are properly wired through to auto-started agents.
+    """
+
+    cron_expression = "*/2 * * * * * *"
+    env_id = UUID(environment)
+
+    result = await client.set_setting(environment, AUTOSTART_AGENT_DEPLOY_INTERVAL, cron_expression)
+    assert result.code == 200
+    result = await client.set_setting(environment, AUTOSTART_AGENT_DEPLOY_SPLAY_TIME, 2)
+    assert result.code == 200
+    result = await client.set_setting(environment, AUTOSTART_AGENT_REPAIR_INTERVAL, cron_expression)
+    assert result.code == 200
```

Comment:
This PR changes the type of the `AUTOSTART_AGENT_DEPLOY_INTERVAL` and `AUTOSTART_AGENT_REPAIR_INTERVAL` settings from int to string. But it's important to remain backwards compatible. As such, we should also test whether the following still works:

Backwards compatibility check with integer values:

* `await client.set_setting(environment, AUTOSTART_AGENT_DEPLOY_INTERVAL, 2)`  -> 
* `result = await client.set_setting(environment, AUTOSTART_AGENT_REPAIR_INTERVAL, 2)`

Test the new datatype (string):

* `await client.set_setting(environment, AUTOSTART_AGENT_DEPLOY_INTERVAL, "2")`  -> 
* `result = await client.set_setting(environment, AUTOSTART_AGENT_REPAIR_INTERVAL, "2")`

Reply:
Updated test case to check that int, string representation of int and cron expression pass validation, and that an empty string fails validation